### PR TITLE
[6.3] FIX UI test_repository search product repo

### PR DESCRIPTION
**Pull request by ldjebran**
repository search was not clicking the search button
```console
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ py.test tests/foreman/ui/test_repository.py -v -k "test_positive_update_custom_ostree_repo_name"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/bin/python2.7
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.16.0, services-1.2.1, mock-1.6.0, cov-2.3.1
collected 59 items 
2017-07-04 18:26:50 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_custom_ostree_repo_name <- robottelo/decorators/__init__.py PASSED

================================================= 58 tests deselected ==================================================
======================================= 1 passed, 58 deselected in 78.25 seconds =======================================
```

### CODE REVIEW

File: robottelo/ui/repository.py [hunk 1]
```diff
@@ -90,6 +90,8 @@ def search(self, element_name):
         """
         self.navigate_to_entity()
         self.assign_value(locators['repo.search'], element_name)
+        prefix = 'kt_' if self.is_katello else ''
```
Comment (oshtaier) [inline robottelo/ui/repository.py]: these statements make sense only for base class. Here we already know that `is_katello = True`

**Comment by ldjebran** [comment]
This PR need more investigation put DO NOT MERGE

**Comment by ldjebran** [comment]
First time run by chance the test pass, but when updated the the code discover the bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1467722
done a workaround
```console
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ py.test tests/foreman/ui/test_repository.py -v -k "test_positive_update_custom_ostree_repo_name"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/bin/python2.7
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.16.0, services-1.2.1, mock-1.6.0, cov-2.3.1
collected 59 items 
2017-07-04 23:47:49 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_custom_ostree_repo_name <- robottelo/decorators/__init__.py PASSED

================================================= 58 tests deselected ==================================================
======================================= 1 passed, 58 deselected in 78.78 seconds =======================================
```

**Comment by sghai** [comment]
ACK

**Comment by oshtaier** [comment]
You changed search method which is a basic one. Please, re-run module or at least CRUD tests

**Comment by ldjebran** [comment]
Run the ui test_repository module 
```console
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ py.test -v tests/foreman/ui/test_repository.py
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/bin/python2.7
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.16.0, services-1.2.1, mock-1.6.0, cov-2.3.1
collected 59 items 
2017-07-05 12:54:41 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_create_with_invalid_download_policy PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_create_with_invalid_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_create_with_same_names <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_delete_puppet_repo_associated_with_cv <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_download_policy_displayed_for_non_yum_repo PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_update_to_invalid_download_policy PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_upload_puppet PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_negative_upload_rpm PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_background_update_to_immediate PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_background_update_to_on_demand PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_custom_ostree_repo <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_immediate_update_to_background PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_immediate_update_to_on_demand PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_in_different_orgs <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_on_demand_update_to_background PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_on_demand_update_to_immediate PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_puppet_repo_same_url_different_orgs PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_repo_with_checksum <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_with_default_download_policy PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_with_download_policy PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_delete <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_delete_custom_ostree_repo <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_discover_repo_via_existing_product <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_discover_repo_via_new_product <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_download_policy_displayed_for_yum_repos PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync_publish_cv PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_drpm_sync_publish_promote_cv PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_list_puppet_modules_with_multiple_repos PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_remove_content_puppet <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_remove_content_rpm <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_reposet_disable FAILED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_resynchronize_puppet_repo <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_resynchronize_rpm_repo <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_srpm_sync <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/robozilla/decorators/__init__.py SKIPPED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_srpm_sync_publish_cv <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/robozilla/decorators/__init__.py SKIPPED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_srpm_sync_publish_promote_cv <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/robozilla/decorators/__init__.py SKIPPED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_sync_custom_repo_docker <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_sync_custom_repo_puppet <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_sync_custom_repo_yum <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_checksum_type <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_custom_ostree_repo_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_custom_ostree_repo_url <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_gpg <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_url <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_upload_puppet PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_upload_rpm <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/robozilla/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_upload_rpm_non_admin <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/robozilla/decorators/__init__.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_local_create <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_local_delete <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_local_update <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_remote_create <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_remote_delete <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_remote_update <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_sync <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_sync_schedule <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_sync_with_content_change <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
tests/foreman/ui/test_repository.py::GitPuppetMirrorTestCase::test_positive_git_view_content <- ../../../.pyenv/versions/2.7.13/envs/sat-6.3/lib/python2.7/site-packages/unittest2/case.py SKIPPED
================================== 7 failed, 38 passed, 14 skipped in 4985.19 seconds ==================================
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ 
```
7 tests failed:
test_positive_create_custom_ostree_repo
test_positive_create_in_different_orgs
test_positive_create_repo_with_checksum
test_positive_delete
test_positive_sync_custom_repo_yum
test_positive_update_url
test_positive_reposet_disable
Fixed:
```console
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ py.test tests/foreman/ui/test_repository.py -v -k "positive_create_custom_ostree_repo or test_positive_create_in_different_orgs or test_positive_create_repo_with_checksum or test_positive_delete or test_positive_sync_custom_repo_yum or test_positive_update_url"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/bin/python2.7
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.16.0, services-1.2.1, mock-1.6.0, cov-2.3.1
collected 59 items 
2017-07-05 17:47:27 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_custom_ostree_repo <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_in_different_orgs <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_create_repo_with_checksum <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_delete_custom_ostree_repo <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_sync_custom_repo_yum <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_repository.py::RepositoryTestCase::test_positive_update_url <- robottelo/decorators/__init__.py PASSED

================================================= 52 tests deselected ==================================================
====================================== 7 passed, 52 deselected in 1804.08 seconds ======================================
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ 
```
in this module remain failing test_positive_reposet_disable with an other behavior and need more investigation if fix needed will be in an other PR

**Pull request merged**